src/Dfe.PlanTech.AzureFunctions/Mappings/EntityUpdater.cs

```diff
@@ -50,4 +50,111 @@ public virtual MappedEntity UpdateEntityConcrete(MappedEntity entity)
+    }
+
+    protected virtual async Task AddOrUpdateReferencedEntities<TEntity, TReferencedEntity>(TEntity entity,
+                                                                                           List<TReferencedEntity> destinationReferenceEntityCollection,
+                                                                                           List<TReferencedEntity> incomingReferencedEntities,
+                                                                                           DbSet<TReferencedEntity> referencedEntityDbSet,
+                                                                                           bool updateOrder = false)
+          where TEntity : ContentComponentDbEntity
+          where TReferencedEntity : ContentComponentDbEntity
+    {
+        int order = 0;
+
+        foreach (var incomingReferencedEntity in incomingReferencedEntities)
+        {
+            var existingReferencedEntity = destinationReferenceEntityCollection.Find(existingReference => existingReference.Id == incomingReferencedEntity.Id) ??
+                                        await AddNewReferencedEntity(entity, incomingReferencedEntity, destinationReferenceEntityCollection, referencedEntityDbSet);
+
+            if (updateOrder && existingReferencedEntity != null)
+            {
+                existingReferencedEntity.Order = order;
+            }
+
+            order++;
+        }
+
+    }
+
+    protected virtual async Task<TReferencedEntity?> AddNewReferencedEntity<TEntity, TReferencedEntity>(TEntity entity, TReferencedEntity referencedEntity, List<TReferencedEntity> referencedEntitiesCollection, DbSet<TReferencedEntity> referencedEntityDbSet)
+        where TEntity : ContentComponentDbEntity
+        where TReferencedEntity : ContentComponentDbEntity
+    {
+        var dbReferencedEntity = await referencedEntityDbSet.FirstOrDefaultAsync(dbReferenceEntity => dbReferenceEntity.Id == referencedEntity.Id);
+        if (dbReferencedEntity == null)
+        {
+            _logger.LogError("{ParentEntityType} {ParentId} is trying to add {ChildReferenceType} {ChildReferenceId} but this is not found in the DB", typeof(TEntity).Name, entity.Id, typeof(TReferencedEntity).Name, referencedEntity.Id);
+            return null;
+        }
+
+        referencedEntitiesCollection.Add(dbReferencedEntity);
+        return dbReferencedEntity;
+    }
+
+    protected virtual void RemoveRemovedReferencedEntities<TReferencedEntity>(List<TReferencedEntity> existingReferencedEntities, List<TReferencedEntity> incomingReferencedEntities)
+        where TReferencedEntity : ContentComponentDbEntity
+    {
+        var referencesToRemove = existingReferencedEntities.Where(existingReference => !incomingReferencedEntities.Exists(incomingReference => incomingReference.Id == existingReference.Id))
+                                                            .ToArray();
```

Comment:
mad new styling or is this Git presenting it to me funny?

Reply:
The fact `.ToArray()` is on another line? Reads better to me, the line is already rather long!